client/messaging_contentstore/tests/test_contactstore.py

```diff
@@ -0,0 +1,227 @@
+        existing_message = make_message_dict(message_data)
+        self.contentstore_backend.messages.endpoint_data[
+            existing_message[u"id"]] = existing_message
+        return existing_message
+
+    def make_existing_schedule(self, schedule_data):
+        existing_schedule = make_schedule_dict(schedule_data)
+        self.contentstore_backend.schedules.endpoint_data[
+            existing_schedule[u"id"]] = existing_schedule
+        return existing_schedule
+
+    def assert_messageset_status(self, messageset_id, exists=True):
+        exists_status = (messageset_id in self.messageset_data)
+        self.assertEqual(exists_status, exists)
+
+    def assert_http_error(self, expected_status, func, *args, **kw):
+        try:
+            func(*args, **kw)
+        except HTTPError as err:
+            self.assertEqual(err.response.status_code, expected_status)
+        else:
+            self.fail(
+                "Expected HTTPError with status %s." % (expected_status,))
+
+    def test_assert_http_error(self):
+        self.session.mount("http://bad.example.com/", TestAdapter("", 500))
+
+        def bad_req():
+            r = self.session.get("http://bad.example.com/")
+            r.raise_for_status()
+
+        # Fails when no exception is raised.
+        self.assertRaises(
+            self.failureException, self.assert_http_error, 404, lambda: None)
+
+        # Fails when an HTTPError with the wrong status code is raised.
+        self.assertRaises(
+            self.failureException, self.assert_http_error, 404, bad_req)
+
+        # Passes when an HTTPError with the expected status code is raised.
+        self.assert_http_error(500, bad_req)
+
+        # Non-HTTPError exceptions aren't caught.
+        def raise_error():
+            raise ValueError()
+
+        self.assertRaises(ValueError, self.assert_http_error, 404, raise_error)
+
+    def test_default_session(self):
+        import requests
+        contentstore = ContentStoreApiClient(self.AUTH_TOKEN)
+        self.assertTrue(isinstance(contentstore.session, requests.Session))
+
+    def test_default_api_url(self):
+        contentstore = ContentStoreApiClient(self.AUTH_TOKEN)
+        self.assertEqual(
+            contentstore.api_url, "http://testserver/contentstore")
+
+    def test_auth_failure(self):
+        contentstore = self.make_client(auth_token="bogus_token")
+        self.assert_http_error(403, contentstore.get_messagesets)
+
+    def test_get_messageset(self):
+        expected_messageset = self.make_existing_messageset({
+            u"short_name": u"Full Set",
+            u"notes": u"A full set of messages.",
+            u"default_schedule": 1
+        })
+        contentstore = self.make_client()
+        [messageset] = list(contentstore.get_messagesets())
+        self.assertEqual(messageset, expected_messageset)
+
+    def test_create_messageset(self):
+        contentstore = self.make_client()
```

Comment:
We could probably have a `self.client = self.make_client()` in `setup`, since we use the client in every test.